frontend/src/pages/modelServing/screens/projects/utils.ts

```diff
@@ -35,28 +35,19 @@ export const isServingRuntimeTokenEnabled = (servingRuntime: ServingRuntimeKind)
 export const isServingRuntimeRouteEnabled = (servingRuntime: ServingRuntimeKind): boolean =>
   servingRuntime.metadata.annotations?.['enable-route'] === 'true';
 
-export const getDefaultServingRuntime = (
-  servingRuntimesConfig?: ConfigMapKind,
-): ServingRuntimeKind => {
-  if (!servingRuntimesConfig) {
-    return DEFAULT_MODEL_SERVING_TEMPLATE;
-  }
-  try {
-    const overrideServingRuntime = YAML.parse(
-      servingRuntimesConfig?.data?.['override-config'] || '',
-    );
-    if (overrideServingRuntime) {
-      return overrideServingRuntime;
-    }
-    const defaultServingRuntime = YAML.parse(servingRuntimesConfig?.data?.['default-config'] || '');
-    if (defaultServingRuntime) {
-      return defaultServingRuntime;
-    }
-  } catch {
-    return DEFAULT_MODEL_SERVING_TEMPLATE;
-  }
-
-  return DEFAULT_MODEL_SERVING_TEMPLATE;
+export const filterTokens = (
```

Comment:
Mmmmm because at the time of fetching data in the context, you don't select any serving runtime, you filter the secrets for each individual serving runtime deployed so you get the tokens (which are the user auth tokens)

Reply:
Yeah, that's what I mean. We can pass the `servingRuntimeName` as the parameter but we can get all the other data in the context, right? So in context provider, we can create a function like
```
const filterTokens = (servingRuntimeName: string): SecretKind[] => {
  const serviceAccountName = customServingRuntimesEnabled
    ? getModelServingPermissionName(servingRuntimeName)
    : getModelServiceAccountName(namespace);
  return secrets?.filter(
    (secret) =>
      secret.metadata.annotations?.['kubernetes.io/service-account.name'] === serviceAccountName,
  );
};
```
Because we can get `customServingRuntimesEnabled`, `namespace` and `secrets` from in the provider (which is the same file).
And then in the component, we only need to get this function from the context and call it, is that possible to implement?